test/unit/tuple_bigref.c

```diff
@@ -5,14 +5,73 @@
 #include <stdio.h>
 
 enum {
-	BIGREF_DIFF = 10,
-	BIGREF_COUNT = 70003,
-	BIGREF_CAPACITY = 107,
+	FEW_REFS = 10,
+	MANY_REFS = 1000,
+	TEST_MAX_TUPLE_COUNT = 1024,
+	RAND_TEST_ROUNDS = 1024 * 1024,
 };
 
 static char tuple_buf[64];
 static char *tuple_end = tuple_buf;
 
+size_t tuple_count = 0;
+static struct tuple *allocated_tuples[TEST_MAX_TUPLE_COUNT];
+
+static struct tuple_format *patched_format;
+static struct tuple * (*save_tuple_new)(struct tuple_format *format,
+					const char *data, const char *end);
+static void (*save_tuple_delete)(struct tuple_format *format,
+				 struct tuple *tuple);
+
+static void patch_format();
+static void restore_format();
+
+static struct tuple *
+test_tuple_new(struct tuple_format *format,
+	       const char *data, const char *end)
+{
+	assert(format == patched_format);
+	assert(tuple_count < TEST_MAX_TUPLE_COUNT);
+	restore_format();
+	struct tuple *tuple = save_tuple_new(format, data, end);
+	patch_format();
+	allocated_tuples[tuple_count++] = tuple;
+	return tuple;
+}
+
+static void
+test_tuple_delete(struct tuple_format *format, struct tuple *tuple)
+{
+	assert(format == patched_format);
+	assert(tuple_count > 0);
+	size_t pos = 0;
+	while (pos < tuple_count && allocated_tuples[pos] != tuple)
+		pos++;
+	ok(pos < tuple_count, "Deleting tuple *was* allocated");
```

Comment:
Tuple deleting what?

Reply:
Deleting is adjective, not the word order. The tuple is deleting and I want to check here that the tuple was allocated before.

Reply:
Then it should say "Tuple being deleted" or "Tuple to be deleted".

Reply:
Got it, fixed.